src/main/java/seedu/dengue/logic/parser/RedoCommandParser.java

```diff
@@ -21,10 +22,21 @@ public RedoCommand parse(String args) throws ParseException {
         }
         try {
             int numberofRedos = Integer.parseInt(trimmedArgs);
+            requirePositive(numberofRedos);
             return new RedoCommand(numberofRedos);
-        } catch (Exception err) {
+        } catch (NumberFormatException err) {
             throw new ParseException(
                     String.format(MESSAGE_INVALID_COMMAND_FORMAT, RedoCommand.MESSAGE_USAGE));
         }
     }
+
+    @Override
+    public void requirePositive(int number) throws ParseException {
+        if (number <= 0) {
+            throw new ParseException(
+                    String.format(
+                            MESSAGE_INVALID_COMMAND_FORMAT
+                                    + MESSAGE_INDICATE_POSITIVE, RedoCommand.MESSAGE_USAGE));
```

Comment:
Just the second message alone might be sufficient, since the command _format_ itself isn't wrong, just the arguments